CAP/gap/UniversalObjects.gd

```diff
@@ -2317,6 +2317,319 @@ DeclareOperation( "AddDirectProductFunctorialWithGivenDirectProducts",
+#! @Returns an object
+#! @Arguments D, method_selection_morphism
+DeclareOperationWithCache( "EqualizerOp",
+                           [ IsList, IsCapCategoryMorphism ] );
+
+#! @Description
+#! The arguments are a list of morphisms $D = ( \beta_i: A \rightarrow B )_{i = 1 \dots n}$.
+#! The output is the equalizer embedding
+#! $\iota: \mathrm{Equalizer}(D) \rightarrow A$.
+#! @Returns a morphism in $\mathrm{Hom}( \mathrm{Equalizer}(D), A )$
+#! @Arguments D
+DeclareOperation( "EmbeddingOfEqualizer",
+                  [ IsList ] );
+
+#! @Description
+#! The arguments are a list of morphisms $D = ( \beta_i: A \rightarrow B )_{i = 1 \dots n}$.
+#! and a morphism for method selection.
+#! The output is the equalizer embedding
+#! $\iota: \mathrm{Equalizer}(D) \rightarrow A$.
+#! @Returns a morphism in $\mathrm{Hom}( \mathrm{Equalizer}(D), A )$
+#! @Arguments D,method_selection_morphism
+DeclareOperation( "EmbeddingOfEqualizerOp",
+                  [ IsList, IsCapCategoryMorphism ] );
+
+#! @Description
+#! The arguments are a list of morphisms $D = ( \beta_i: A \rightarrow B )_{i = 1 \dots n}$,
+#! and an object $E = \mathrm{Equalizer}(D)$.
+#! The output is the equalizer embedding
+#! $\iota: E \rightarrow A$.
+#! @Returns a morphism in $\mathrm{Hom}( E, A )$
+#! @Arguments D,E
+DeclareOperation( "EmbeddingOfEqualizerWithGivenEqualizer",
+                  [ IsList, IsCapCategoryObject ] );
+
+#! @Description
+#! The arguments are a list of morphisms $D = ( \beta_i: A \rightarrow B )_{i = 1 \dots n}$
+#! and a morphism $ \tau: T \rightarrow A $
+#! such that $\beta_i \circ \tau  \sim_{T, B} \beta_j \circ \tau$ for all pairs $i,j$.
+#! The output is the morphism
+#! $u( \tau ): T \rightarrow \mathrm{Equalizer}(D)$
+#! given by the universal property of the equalizer.
+#! @Returns a morphism in $\mathrm{Hom}( T, \mathrm{Equalizer}(D) )$
+#! @Arguments D, tau
+DeclareOperation( "UniversalMorphismIntoEqualizer",
+                  [ IsList, IsCapCategoryMorphism ] );
+
+#! @Description
+#! The arguments are a list of morphisms $D = ( \beta_i: A \rightarrow B )_{i = 1 \dots n}$,
+#! a morphism $\tau: T \rightarrow A )$
+#! such that $\beta_i \circ \tau  \sim_{T, B} \beta_j \circ \tau$ for all pairs $i,j$,
+#! and an object $E = \mathrm{Equalizer}(D)$.
+#! The output is the morphism
+#! $u( \tau ): T \rightarrow E$
+#! given by the universal property of the equalizer.
+#! @Returns a morphism in $\mathrm{Hom}( T, E )$
+#! @Arguments D, tau, E
+DeclareOperation( "UniversalMorphismIntoEqualizerWithGivenEqualizer",
+                  [ IsList, IsCapCategoryMorphism, IsCapCategoryObject ] );
+
+## Add Operations
+
+#! @Description
+#! The arguments are a category $C$ and a function $F$.
+#! This operations adds the given function $F$
+#! to the category for the basic operation <C>Equalizer</C>.
+#! $F: ( (\beta_i: A \rightarrow B)_{i = 1 \dots n} ) \mapsto E$
+#! @Returns nothing
+#! @Arguments C, F
+DeclareOperation( "AddEqualizer",
```

Comment:
Should we group those for the documentation?

Reply:
I would propose to do this in a follow-up since similar pieces of code already exist in CAP.

Reply:
All right.